crates/l2/prover/zkvm/interface/src/io.rs

```diff
@@ -114,3 +119,82 @@ impl ProgramOutput {
         .concat()
     }
 }
+
+#[derive(Clone)]
+pub struct ExecutionWitnessWrapper {
+    inner: Arc<Mutex<ExecutionWitnessResult>>,
+}
+
+impl ExecutionWitnessWrapper {
+    pub fn new(db: ExecutionWitnessResult) -> Self {
+        Self {
+            inner: Arc::new(Mutex::new(db)),
+        }
+    }
+
+    pub fn lock_mutex(&self) -> Result<MutexGuard<ExecutionWitnessResult>, ExecutionWitnessError> {
```

Comment:
You can remove this since it is not being called outside the impl block
```suggestion
    fn lock_mutex(&self) -> Result<MutexGuard<ExecutionWitnessResult>, ExecutionWitnessError> {
```

Reply:
done while moving everything to witness_db.rs [here](https://github.com/lambdaclass/ethrex/pull/3765/commits/c0b4b412406fc785014ce07e36ddfb10847812a2)